src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,235 @@
+package duke;
+
+import duke.exception.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Task;
+import duke.task.Event;
+import duke.task.ToDo;
+import duke.task.Deadline;
+
+
+public class TaskList {
+
+	private final ArrayList<Task> tasks = new ArrayList<>();
+
+	/**
+	 * Add task to list.
+	 */
+	public void addTask(Task task) {
+		tasks.add(task);
+	}
+
+	/**
+	 * Deletes the task from list.
+	 *
+	 * @param task task
+	 */
+	private void deleteTask(Task task) {
+		tasks.remove(task);
+	}
+
+	/**
+	 * Marks the task as complete and prints out the task, throws error if done command is not properly formatted.
+	 *
+	 * @param userInput the index of the task in lis.
+	 */
+	public void completeAndPrintTask(String userInput) {
+		String[] tokens = userInput.split(" ");
+		int taskNumber;
+		boolean isTaskDoneCommand = tokens.length == 2 && tokens[0].equals("done");
+		try {
+			try {
+				taskNumber = Integer.parseInt(tokens[1]);
+			} catch (NumberFormatException err) {
+				throw new InvalidIndexException("done");
+			}
+			if (!isTaskDoneCommand || taskNumber > tasks.size()) {
+				throw new InvalidIndexException("done");
+			}
+		} catch (InvalidIndexException err) {
+			UI.printErrorMessage(err);
+			return;
+		}
```

Comment:
Generally exception as control flow (i.e. throwing an exception above and catching it a few lines below) is not a good idea. Maybe a better way for this try-catch block might be:  
```
try {
    taskNumber = Integer.parseInt(tokens[1]);
    if (!isTaskDoneCommand || taskNumber > tasks.size()) {
        throw new InvalidIndexException("done");
    }
catch (NumberFormatException err) {
     throw new InvalidIndexException("done");
}
```
and catch `InvalidIndexException` in `src/main/java/Duke/Parser.java`.